suitcase/spec.py

```diff
@@ -1074,3 +1282,151 @@ def stop(self, doc):
+                                check_in_broker=True):
+
+    """
+    Insert all (most) scans in a specfile into the databroker
+
+    Scans whose type is not in the module-level variable _BLUESKY_PLAN_NAMES
+    will not be inserted into the databroker.
+
+    Parameters
+    ----------
+    specfile : Specfile
+        The Specfile object to insert into the databroker
+    validate : bool
+        Defaults to False.
+        True/False: Do/Don't validate the document dict against the json
+                    schema defined in event_model
+    check_in_broker : bool
+        Defaults to True.
+        True/False: Do/Don't check to see if the document already exists in
+                    metadatastore.  If it does, the document will be replaced
+                    with the one that is already in metadatastore
+
+    Returns
+    -------
+    suceeded : list
+        List of tuples of the scan object and a list of ``_inserted`` namedtuples
+        that contain the document name, the document uid and whether or not
+        the insertion was successful for every document that was inserted into
+        metadatastore
+    failed : list
+        List of tuples of the scan object and the error that was raised for
+        any scans that failed to insert
+
+    Notes
+    -----
+    see `summarize_insertion` for a helper function to format the return values
+    of this function
+    """
+    suceeded = []
+    failed = []
+    for scan in specfile:
+        try:
+            uids = insert_specscan_into_broker(scan, validate=validate,
+                                               check_in_broker=check_in_broker)
+            suceeded.append((scan, uids))
+        except NotImplementedError as e:
+            failed.append((scan, e))
+
+    return suceeded, failed
+
+
+def summarize_insertion(suceeded, failed):
+    """
+    Helper function that takes the output of `insert_specfile_into_broker`,
+    formats it into easy-to-parse information and prints it out to the console
+
+    Usage
+    -----
+    >>> ret = insert_specfile_into_broker(specfile_object)
+    >>> summarize_insertion(*ret)
+    """
+    logger.info('{} scans failed to be inserted'.format(len(failed)))
+    for scan, exception in failed:
+        logger.info('{:>10}: {}'.format(scan.scan_id, exception))
+    order = [event_model.DocumentNames.start,
+             event_model.DocumentNames.descriptor,
+             event_model.DocumentNames.event,
+             event_model.DocumentNames.stop]
+    table = PrettyTable(field_names=['scan_ids', 'start', 'descriptor',
+                                     'event', 'stop'])
+    for scan, uids in suceeded:
+        docs = defaultdict(list)
+        for uid in uids:
+            docs[uid.doc_name].append(uid)
+        # print out a summary that shows the number of documents that were
+        # inserted for each scan, including the number that already existed in
+        # metadatastore
+        summarize = lambda x: "{}/{}".format(
```

Comment:
> my linter will whine about this

I can't tell if you are going to make me turn this into a function or not. Could you just say what you want instead of me having to guess?